src/gradualizer_db.erl

```diff
@@ -279,17 +280,29 @@ autoimport(M, #state{opts = #{autoimport := true},
     end.
 
 import_module(Mod, State) ->
-    case State#state.srcmap of
+    case State#state.beammap of
         #{Mod := Filename} ->
             State1 = import_files([Filename], State),
             {ok, State1};
         _ ->
-            not_found
+          case State#state.srcmap of
+              #{Mod := Filename} ->
+                  State1 = import_files([Filename], State),
+                  {ok, State1};
+              _ ->
+                  not_found
+          end
     end.
 
 import_files([File | Files], State) ->
-    EppOpts = [{includes, guess_include_dirs(File)}],
-    {ok, Forms} = epp:parse_file(File, EppOpts),
+    {ok, Forms} =
+        case re:run(File, beam_file_regexp()) of
+            {match, _} ->
+                {ok, gradualizer:get_forms_from_beam(File)};
+            nomatch ->
+                EppOpts = [{includes, guess_include_dirs(File)}],
+                epp:parse_file(File, EppOpts)
+        end,
     [{attribute, _, file, _},
      {attribute, _, module, Module} | Forms1] = Forms,
     check_epp_errors(File, Forms1),
```

Comment:
Check_epp_errors should probably be done only when we use epp, i.e. not for beam files.